delta-app/src/main/java/io/cdap/delta/app/DeltaContext.java

```diff
@@ -73,18 +75,31 @@
 
   @Override
   public void incrementCount(DMLOperation op) {
-    eventMetrics.incrementDMLCount(op);
+    String tableName = op.getTableName();
+    EventMetrics eventMetrics = tableEventMetrics.computeIfAbsent(
+      tableName, s -> new EventMetrics(metrics.child(ImmutableMap.of(PROGRAM_METRIC_ENTITY, tableName))));
+    eventMetrics.incrementDMLCount(op.getType());
   }
 
   @Override
   public void incrementCount(DDLOperation op) {
+    String tableName = op.getTableName();
+    if (tableName == null) {
+      // This can happen for DDL operations such as CREATE_DATABASE
+      return;
+    }
+    EventMetrics eventMetrics = tableEventMetrics.computeIfAbsent(
+      tableName, s -> new EventMetrics(metrics.child(ImmutableMap.of(PROGRAM_METRIC_ENTITY, tableName))));
     eventMetrics.incrementDDLCount();
   }
 
   @Override
   public void commitOffset(Offset offset, long sequenceNumber) throws IOException {
     stateStore.writeOffset(id, new OffsetAndSequence(offset, sequenceNumber));
-    eventMetrics.emitMetrics();
+    for (EventMetrics eventMetrics : tableEventMetrics.values()) {
+      eventMetrics.emitMetrics();
+    }
+    tableEventMetrics.clear();
```

Comment:
nit: shall we call clearMetrics() here ? so that if you have implementation changes in clearMetrics, it will be auto applied here.

Reply:
makes sense 👍 fixed.